hail/python/hail/ir/ir.py

```diff
@@ -729,14 +729,21 @@ def renderable_bindings(self, i, default_value=None):
 
 
 class NDArrayRef(IR):
-    @typecheck_method(nd=IR, idxs=sequenceof(IR))
-    def __init__(self, nd, idxs):
+    @typecheck_method(nd=IR, idxs=sequenceof(IR), error_id=nullable(int), stack_trace=nullable(str))
+    def __init__(self, nd, idxs, error_id=None, stack_trace=None):
         super().__init__(nd, *idxs)
         self.nd = nd
         self.idxs = idxs
+        self._error_id = error_id
+        self._stack_trace = stack_trace
+        if error_id is None or stack_trace is None:
+            self.save_error_info()
```

Comment:
I'm not happy about having to change the constructor in Python like this. Ideally, we'd just let `self.save_error_info` in the constructor do all the work. However, when we copy IR around, we need to copy the error id and stack trace too. Open to suggestions about better ways to do this.